docs/source/format/DissociatedIPC.rst

```diff
@@ -0,0 +1,335 @@
+* Tagged message sending
+  - Sending a message that has an attached little-endian, unsigned 64-bit integral tag
+    for control flow. A tag like this allows control flow to operate on a message whose body
+    is on a non-CPU device without requiring the message itself to get copied off of the device.
+
+URI Specification
+-----------------
+
+When providing a URI to a consumer to contact for use with this protocol (such as via 
+the `Location URI for Flight <flight-location-uris>`_), the URI should specify a scheme
+like *ucx://* or *fabric://*, that is easily identifiable. In addition, the URI should
+encode the following URI query parameters:
+
+.. note::
+    As this protocol matures, this document will get updated with commonly recognized
+    transport schemes that get used with it.
+
+* ``want_data`` - **REQUIRED** - uint64 integer value
+  - This value should be used to tag an initial message to the server to initiate a
+    data transfer. The body of the initiating message should be an opaque binary identifier
+    of the data stream being requested (like the ``Ticket`` in the Flight RPC protocol)
+* ``free_data`` - **OPTIONAL** - uint64 integer value
+  - If the server might send messages using offsets / addresses for remote memory accessing
+    or shared memory locations, the URI should include this parameter. This value is used to
+    tag messages sent from the client to the data server, containing specific offsets / addresses
+    which were provided that are no longer required by the client (i.e. any operations that
+    directly reference those memory locations, such as copying the remote data into local memory,
+    have been completed).
+* ``remote_handle`` - **OPTIONAL** - base64-encoded string
+  - When working with shared memory or remote memory, this value indicates any required
+    handle or identifier that is necessary for accessing the memory.
+    + Using UCX, this would be an *rkey* value
+    + With CUDA IPC, this would be the value of the base GPU pointer or memory handle,
+      and subsequent addresses would be offsets from this base pointer.
+
+Protocol Description
+====================
+
+There are two possibilities that can occur:
+
+1. The streams of metadata and body data are across separate connections
+
+.. figure:: ./DissociatedIPC/SequenceDiagramSeparate.mmd.svg
+
+2. The streams of metadata and body data are sent simultaneously across the
+   same connection
+
+.. figure:: ./DissociatedIPC/SequenceDiagramSame.mmd.svg
+
+Server Sequence
+---------------
+
+There can be either a single server handling both the IPC Metadata stream and the
+Body data streams, or separate servers for handling the IPC Metadata and the body
+data. This allows for streaming of data across either a single transport pipe or
+two pipes if desired.
+
+Metadata Stream Sequence
+''''''''''''''''''''''''
+
+The standing state of the server is waiting for a **tagged** message with the specified
```

Comment:
Specified by whom? This isn't clear.

Reply:
I'll make this more clear. Ideally the server specifies its `want_data` tag, and then just needs to make sure the value is propagated however the URI is being sent to the client.